src/storage.ts

```diff
@@ -0,0 +1,78 @@
+import { FastifyLoggerInstance } from 'fastify';
+
+
+const cloneDeep = (item: any[]) => JSON.parse(JSON.stringify(item));
+
+
+export interface IStorage {
+    readonly logger: FastifyLoggerInstance,
+    add: (data: any) => object | boolean,
+    read: (id?: string) => object[],
+    update: (id: string, data: any) => object | boolean,
+    delete: (id: string) => boolean
+}
+
+export class Storage implements IStorage {
+    readonly logger: FastifyLoggerInstance;
+    private __store!: {[key: string]: any}[];
+
+    constructor ( logger: FastifyLoggerInstance ) {
+        this.logger = logger;
+        Object.defineProperty(this, '__store', {
```

Comment:
why defineProperty?

Reply:
Destructive actions outside the class like:

`delete storage.__store;`
`storage.__store = null;`
etc.

of intentional or unintentional nature, are failed, silently of with such error `TypeError: Cannot assign to read only property '__store' of object '#<Storage>'`.

With some TS flags it's possible to use `#` for truly privateness.

Thank you, I mistakenly added a `writable` descriptor.